src/Network/Broadcast/OutboundQueue.hs

```diff
@@ -1005,19 +1218,32 @@ enqueueCherished outQ msgType msg =
 -------------------------------------------------------------------------------}
 
 -- | Enqueue message to the specified set of peers
-intEnqueueTo :: (MonadIO m, WithLogger m)
+intEnqueueTo :: forall m msg nid buck a. (MonadIO m, WithLogger m)
              => OutboundQ msg nid buck
              -> MsgType nid
              -> msg a
              -> EnqueueTo nid
              -> m [Packet msg nid a]
-intEnqueueTo outQ@OutQ{} msgType msg enqTo = do
-    peers <- getAllPeers outQ
-    intEnqueue outQ msgType msg (restriction peers)
+intEnqueueTo outQ@OutQ{..} msgType msg enqTo = do
+    peers <- restrict =<< getAllPeers outQ
+    intEnqueue outQ msgType msg peers
   where
-    restriction = case enqTo of
-      EnqueueToAll -> id
-      EnqueueToSubset peers' -> restrictPeers peers'
+    -- Restrict the set of all peers to the requested subset (if one).
+    --
+    -- Any unknown peers will be used as a fallback in case no other peers
+    -- can be found.
```

Comment:
The relays _will_ send to unknown peers because the static topology will not be bidirectional.